### PR TITLE
Get config from server

### PR DESCRIPTION
**Pull request by Nokitoo**
Changed Mage SDK setup process:
- No longer need to feed Webpack with mage.getClientConfig
- Add Mage.setEndpoint to setup server endpoint
- add Mage.init to fetch config from the Mage server

Fixes mage/mage#239

### CODE REVIEW

File: docs/ReadMe.md [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
 
-- You can write the configuration to a JSON file that your build system picks up and embeds.
-- You can host the JSON on an HTTP endpoint.
-- You can have both client-code and server-code in a single repository, and integrate the build system into it.
 
-Which approach you take will depend on your desired workflow. In the [integrations](./integrations) folder we provide
-examples for specific build systems.
+## Exposing the client config
+
+To init the sdk, your MAGE server needs to expose the client config.
```
Comment (stelcheck) [inline docs/ReadMe.md]: SDK, not sdk.

File: docs/ReadMe.md [hunk 2]
```diff
@@ -47,26 +47,47 @@ up yet.
 ```javascript
 var mage = require('mage-sdk-js');
 
-// register and set up all modules that don't require login
+mage.setEndpoint(
+	'http://127.0.0.1', // Base url
+	'game' 				// App name
```
Comment (stelcheck) [inline docs/ReadMe.md]: We could make this optional (default: 'game')

File: docs/ReadMe.md [hunk 3]
```diff
@@ -47,26 +47,47 @@ up yet.
...
+);
+
+// Init mage sdk
```
Comment (stelcheck) [inline docs/ReadMe.md]: SDK

File: docs/ReadMe.md [hunk 4]
```diff
@@ -47,26 +47,47 @@ up yet.
...
+// It will fetch the usercommands config from the endpoint
+
+mage.init(function(err) {
+	if (err) {
+		return;
```
Comment (stelcheck) [inline docs/ReadMe.md]: Should at least log

File: lib/Mage.js [hunk 5]
```diff
@@ -3,22 +3,67 @@ var inherits = require('inherits');
...
 
 module.exports = Mage;
 
+// call config.get usercommand to get client config and init mage sdk with it
+//
+// the usercommand is called manually because CommandCenter needs the config to setup
+
+Mage.prototype.init = function (cb) {
+	if (!this.endpoint) {
+		throw new Error('Please call Mage.setEndpoint before calling Mage.init');
```
Comment (stelcheck) [inline lib/Mage.js]: mage.setInstance; `Mage...` looks like a static method.

File: lib/Mage.js [hunk 6]
```diff
@@ -3,22 +3,67 @@ var inherits = require('inherits');
...
+	}
+
+	var that = this;
+	var hr = new HttpRequest({
```
Comment (stelcheck) [inline lib/Mage.js]: I think we discussed this, but can't we use `sendCommand` somehow? For the record, why couldn't we use it?

**Review by stelcheck — CHANGES_REQUESTED** [review]
Overall it goes in the right direction.

File: lib/Mage.js (hunk 6 above)
Comment (Nokitoo) [reply lib/Mage.js]: Because the `CommandCenter` need the config to setup.

**Comment by Nokitoo** [comment]
I don't know if it should be in this PR or a new one, but I think we should definitely remove all client related config from the server config.
For example, `server.msgStream.detect` is only used by the SDK to know in what order it should attempt to create protocols.

**Comment by stelcheck** [comment]
We can do this in another PR, yes.

**Review by stelcheck — APPROVED** [review]

**Pull request merged**